docs/howto/handling_connections.ipynb

```diff
@@ -4,16 +4,176 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# How to handle connections to devices\n",
+    "# How to handle connections to devices"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Connection options\n",
+    "By default, connections are handled automatically based on the information in the inventory.\n",
+    "Here is an example of *hosts.yaml* with simple and advanced connection details:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/markdown": [
+       "```yaml\n",
+       "simple_host:\n",
+       "    hostname: simple-host.my-lab.com\n",
+       "    username: simple-user\n",
+       "    password: simple-password\n",
+       "    platform: cisco_ios\n",
+       "\n",
+       "advanced_host:\n",
+       "    # standard parameters that will be re-used across different connections\n",
+       "    hostname: advanced-host.my-lab.com\n",
+       "    username: my_user\n",
+       "    password: my_password\n",
+       "    platform: linux\n",
+       "    connection_options:\n",
+       "        napalm:\n",
+       "          # standard parameters override upper level and will be used only for the napalm connection\n",
+       "          port: 443\n",
+       "          platform: eos\n",
+       "          extras:\n",
+       "              # non-standard library-specific options (check backend library docs)\n",
+       "              optional_args:\n",
+       "                  eos_autoComplete: true\n",
+       "        netmiko:\n",
+       "            platform: arista_eos\n",
+       "            extras:\n",
+       "                secret: \"my-enable-password\"\n",
+       "                global_delay_factor: 2\n",
+       "        netmiko_telnet:  # non-default connection name\n",
```

Comment:
how do you know this is a netmiko connection?

Reply:
I don't know beforehand, but I deduce it using the task. E.g. `netmiko_send_command` uses plugin netmiko so if we use a connection named `netmiko_telnet` for `netmiko_send_command` task, the connection should be created using netmiko plugin.
Obviously, if you open connection manually, you need to specify it yourself.